pinecone/index.py

```diff
@@ -151,12 +167,68 @@ def _upsert_batch(self,
 
         args_dict = self._parse_non_empty_args([('namespace', namespace)])
```

Comment:
nit : I dont think the parse_non_empty_args function is too necessary, we are not doing any extra validation there